cirq-web/cirq_ts/src/circuit/components/types_test.ts

```diff
@@ -0,0 +1,181 @@
+      },
+    ];
+
+    it('handles the X gate special case', () => {
+      const symbolObj = new Symbol3D(symbols[0]);
+      const xSymbol = symbolObj.children[0];
+      expect(xSymbol instanceof X3DSymbol).to.equal(true);
+    });
+
+    it('handles the control symbol special case', () => {
+      const symbolObj = new Symbol3D(symbols[1]);
+      const ctrlSymbol = symbolObj.children[0];
+      expect(ctrlSymbol instanceof Control3DSymbol).to.equal(true);
+    });
+
+    it('handles an arbitrary symbol with a box', () => {
+      const symbolObj = new Symbol3D(symbols[2]);
+      const boxSymbol = symbolObj.children[0];
+      expect(boxSymbol instanceof BoxGate3DSymbol).to.equal(true);
+    });
+
+    it('builds every symbol in the correct location on the scene', () => {
+      const expectedRows = [0, 1, 3];
+      const expectedCols = [0, 2, 2];
+      const expectedMoments = [1, 3, 7];
+      symbols.forEach((value, index) => {
+        const symbol = new Symbol3D(value).children[0];
+        expect(symbol.position.x).to.equal(expectedRows[index]);
+        expect(symbol.position.y).to.equal(expectedMoments[index]);
+        expect(symbol.position.z).to.equal(expectedCols[index]);
+      });
+    });
+  });
+
+  describe('with valid multi-qubit SymbolInformation objects', () => {
+    const symbols: SymbolInformation[] = [
+      {
+        wire_symbols: ['@', 'X'],
+        location_info: [
+          {row: 0, col: 0},
+          {row: 0, col: 1},
+        ],
+        color_info: ['black', 'black'],
+        moment: 1,
+      },
+      {
+        wire_symbols: ['iSWAP', 'iSWAP'],
+        location_info: [
+          {row: 3, col: 2},
+          {row: 3, col: 3},
+        ],
+        color_info: ['#d3d3d3', '#d3d3d3'],
+        moment: 7,
+      },
+      {
+        wire_symbols: ['@', '@', 'X'],
+        location_info: [
+          {row: 0, col: 0},
+          {row: 0, col: 1},
+          {row: 0, col: 2},
+        ],
+        color_info: ['black', 'black', 'black'],
+        moment: 1,
+      },
+    ];
+
+    it('handles a control symbol and an X gate (CNOT)', () => {
+      const symbolObj = new Symbol3D(symbols[0]);
+
+      const ctrlSymbol = symbolObj.children.find(
+        child => child.constructor.name === 'Control3DSymbol'
+      );
+      expect(ctrlSymbol).to.not.equal(undefined);
+      expect(ctrlSymbol?.position.x).to.equal(0);
+      expect(ctrlSymbol?.position.z).to.equal(0);
+      expect(ctrlSymbol?.position.y).to.equal(1);
+
+      const xSymbol = symbolObj.children.find(
+        child => child.constructor.name === 'X3DSymbol'
+      );
+      expect(xSymbol).to.not.equal(undefined);
+      expect(xSymbol?.position.x).to.equal(0);
+      expect(xSymbol?.position.z).to.equal(1);
+      expect(xSymbol?.position.y).to.equal(1);
+
+      const connectionLine = symbolObj.children.find(
+        child => child.constructor.name === 'ConnectionLine'
+      );
+
+      // TODO: find best way to get position of each line.
```

Comment:
create an issue and reference it in the TODO or do the TODO

Reply:
For testing here I actually don't think it adds anything to get the positions, since we know that connection line already accepts coords as its input. Making sure it is actually at the position would belong in a theoreticaly meshes_test.ts imo